sympy/solvers/solveset.py

```diff
@@ -186,12 +184,27 @@ def inv(trig):
                 if isinstance(f, (tan, cot)):
                     return (lambda a: n*pi + f.inverse()(a),)
 
+        if isinstance(g_ys, FiniteSet):
+            n = Dummy('n', integer=True)
+            invs = S.EmptySet
+            for L in inv(f):
+                invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
+            return _invert_real(f.args[0], invs, symbol)
+        if isinstance(g_ys, ImageSet):
```

Comment:
If we have nested Trig function.
Changes for https://github.com/sympy/sympy/issues/10217